backend/service/topology/querybuilder.go

```diff
@@ -0,0 +1,115 @@
+package topology
+
+import (
+	"fmt"
+	"strings"
+
+	sq "github.com/Masterminds/squirrel"
+
+	topologyv1 "github.com/lyft/clutch/backend/api/topology/v1"
+)
+
+const (
+	column   = "column."
+	metadata = "metadata."
+)
+
+func paginatedQueryBuilder(
+	filter *topologyv1.SearchTopologyRequest_Filter,
+	sort *topologyv1.SearchTopologyRequest_Sort,
+	pageToken,
+	limit int,
+) (sq.SelectBuilder, error) {
+	queryLimit := 100
+	if limit > 0 {
+		queryLimit = limit
```

Comment:
What do you think about having an upper bound for the limit?

Reply:
Yeah totally sounds good, i'll say 1000 for now?